src/plugins/intel_cpu/src/nodes/tensoriterator.cpp

```diff
@@ -568,6 +579,41 @@ void TensorIterator::prepareParams() {
 void TensorIterator::execute(dnnl::stream strm) {
     //Special case, the subgraph is dynamic while the node has all static shapes
     if (runAsDynamic()) {
+        // redefine memory for input nodes of subgraph and
+        // reset first_mappers as the primitives are invalid.
+        auto restoreSubgraphInputByBackEdges = [&]() {
```

Comment:
Please extract this lambda to a separate private method.

Reply:
applied review comments.